class.jetpack.php

```diff
@@ -470,6 +470,10 @@ private function Jetpack() {
 		add_action( 'user_register', array( $this, 'is_single_user_site_invalidate' ) );
 		add_action( 'deleted_user', array( $this, 'is_single_user_site_invalidate' ) );
 
+		// Unlink user before deleting the user from .com
+		add_action( 'deleted_user', array( $this, 'unlink_user' ), 10, 1 );
```

Comment:
`user_role_change()` might help here - especially if the user you're deleting is the master user.